media/js/firefox/new/experiment-search-retention-funnelcakes.js

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function() {
+    'use strict';
+
+    var experimentId = 'experiment_search_retention_funnelcake';
+
+    // swiped from mozilla-client.js
+    var ua = navigator.userAgent;
+    var isLikeFirefox = /Iceweasel|IceCat|SeaMonkey|Camino|like\ Firefox/i.test(ua);
+    var isFirefox = /\s(Firefox|FxiOS)/.test(ua) && !isLikeFirefox(ua);
+    var isMobile = /^(android|ios|fxos)$/.test(window.site.platform);
+
+    var hasCookie = Mozilla.Cookies.hasItem(experimentId);
+    var fcRe = /^.*\?.*f=\d{3}.*/;
+
+    // 0. if visitor already has a cookie for this experiment, skip the expenisve checks below
+    if (hasCookie) {
+        runExperiment();
+    }
+    // 1. check to make sure user is on windows and not fx
+    // 1.1 extra check around DNT so we don't do unnecessary geolookups
+    // 1.2 make sure user isn't on a variation
+    else if (!fcRe.test(document.location.search) && window.site.platform === 'windows' && !isFirefox && !isMobile && typeof Mozilla.dntEnabled === 'function' && !Mozilla.dntEnabled()) {
```

Comment:
This is a bit of a long-winded conditional. Could we use some variables to make it shorter and a bit more readable?